frontend/components/details/ModuleDetail.tsx

```diff
@@ -1,13 +1,30 @@
-import { EntityModule, OccurrenceModule, QuoteModule, TopicModule } from "@/interfaces";
+import PostulationDetail from "@/components/details/PostulationDetail";
+import {
+  EntityModule,
+  ImageModule,
+  OccurrenceModule,
+  PostulationModule,
+  QuoteModule,
+  SourceModule,
+  TopicModule,
+} from "@/interfaces";
 import { useSession } from "next-auth/client";
 import { createRef, FC, useLayoutEffect } from "react";
 import EntityDetail from "./EntityDetail";
+import ImageDetail from "./ImageDetail";
 import OccurrenceDetail from "./OccurrenceDetail";
 import QuoteDetail from "./QuoteDetail";
+import SourceDetail from "./SourceDetail";
 import TopicDetail from "./TopicDetail";
 
 interface ModuleDetailProps {
-  module: OccurrenceModule | QuoteModule | EntityModule | TopicModule;
+  module:
+    | OccurrenceModule
+    | QuoteModule
+    | EntityModule
+    | TopicModule
+    | PostulationModule
+    | ImageModule;
```

Comment:
We'll probably want to add
```ts
export type ModuleUnion = ... | ... | ...
```
to `interfaces.d.ts` when `ModuleUnionCard` gets converted to TS.